runner/generate_helper_sql.go

```diff
@@ -68,3 +68,36 @@ func GenerateStatsHelperSql(ctx context.Context, server *state.Server, globalCol
 
 	return output.String(), nil
 }
+
+func GenerateExplainAnalyzeHelperSql(ctx context.Context, server *state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) (string, error) {
+	db, err := postgres.EstablishConnection(ctx, server, logger, globalCollectionOpts, "")
+	if err != nil {
+		return "", err
+	}
+	defer db.Close()
+
+	version, err := postgres.GetPostgresVersion(ctx, logger, db)
+	if err != nil {
+		return "", fmt.Errorf("error collecting Postgres version: %s", err)
+	}
+
+	databases, _, err := postgres.GetDatabases(ctx, logger, db, version)
+	if err != nil {
+		return "", fmt.Errorf("error collecting pg_databases: %s", err)
+	}
+
+	output := strings.Builder{}
+	for _, dbName := range postgres.GetDatabasesToCollect(server, databases) {
+		output.WriteString(fmt.Sprintf("\\c %s\n", pq.QuoteIdentifier(dbName)))
+		output.WriteString("CREATE SCHEMA IF NOT EXISTS pganalyze;\n")
+		output.WriteString(fmt.Sprintf("GRANT USAGE ON SCHEMA pganalyze TO %s;\n", server.Config.GetDbUsername()))
```

Comment:
Aren't we connecting as `server.Config.GetDbUsername()`? Should this be `globalCollectionOpts.GenerateExplainAnalyzeHelperRole`?

Reply:
Also, technically users may select a helper role name that needs quoting as well, right? But it's probably fine not to support that.

Reply:
> Aren't we connecting as `server.Config.GetDbUsername()`? Should this be `globalCollectionOpts.GenerateExplainAnalyzeHelperRole`?

This `GRANT USAGE` is not for the creation step, but rather so that the regular pganalyze monitoring user can use the function later - so using `server.Config.GetDbUsername()` is correct here.

Note that the generate logic doesn't connect itself, but emits a psql script that you execute as a superuser - so this does not refer to the user that the statement runs as.

Reply:
> Also, technically users may select a helper role name that needs quoting as well, right? But it's probably fine not to support that.

Good point - I can wrap this, and the other uses of the role name in `pq.QuoteIdentifier`.

Reply:
Ah, yes, okay. That makes sense.